src/operator/deconvolution-inl.h

```diff
@@ -43,26 +43,30 @@ struct DeconvolutionParam : public dmlc::Parameter<DeconvolutionParam> {
   bool cudnn_off;
   dmlc::optional<int> layout;
   DMLC_DECLARE_PARAMETER(DeconvolutionParam) {
-    DMLC_DECLARE_FIELD(kernel).describe("deconvolution kernel size: (h, w) or (d, h, w)");
+    DMLC_DECLARE_FIELD(kernel).describe("Deconvolution kernel size: (h, w) or (d, h, w). "
+                  "This is same as the kernel size used for the corresponding convolution");
     DMLC_DECLARE_FIELD(stride).set_default(TShape())
-        .describe("deconvolution stride: (h, w) or (d, h, w)");
+        .describe("The stride used for the corresponding convolution: (h, w) or (d, h, w).");
     DMLC_DECLARE_FIELD(dilate).set_default(TShape())
-    .describe("deconvolution dilate: (h, w) or (d, h, w)");
+        .describe("Dilation factor for each dimension of the input: (h, w) or (d, h, w).");
     DMLC_DECLARE_FIELD(pad).set_default(TShape())
-        .describe("pad for deconvolution: (h, w) or (d, h, w). "
-                  "A good number is : (kernel-1)/2. "
-                  "If target_shape is set, "
-                  "pad will be ignored and computed accordingly");
+        .describe("The amount of implicit zero padding added during convolution for each "
+                  "dimension of the input: "
```

Comment:
The description says this is the padding that was used for the convolution input (not deconvolution input). Also, saying deconvolution pads the output could be misleading. Deconvolution doesn't actually set the borders to zero. Example:
```
>>> # The transpose of convolving a 4×4 kernel over a 5×5 input padded with a
... # 2 × 2 border of zeros using unit strides (i.e., i = 5, k = 4, s = 1 and p = 2).
... input_5x5 = mx.nd.normal(shape=[1, 16, 5, 5])
>>> kernel_4x4 = mx.nd.normal(shape=[16, 16, 4, 4])
>>> conv = mx.nd.Convolution(data=input_5x5, kernel=(4,4), weight=kernel_4x4, pad=(2,2), num_filter=16, no_bias=True)
>>> transpose = mx.nd.Deconvolution(data=conv, kernel=(4,4), weight=kernel_4x4, pad=(2,2), num_filter=16, no_bias=True)
>>> transpose[0][0].asnumpy()
array([[  13.21707153, -174.19221497,  295.35494995, -482.07144165,
         236.75886536],
       [-166.75314331,  161.54397583,  344.41635132, -123.3405838 ,
         -20.51438904],
       [-991.8168335 ,   14.92340851, -131.36260986, -170.46170044,
        -283.42138672],
       [ 165.29977417,  -47.57701492, -143.96827698, -194.85569763,
         -91.3994751 ],
       [   1.36075592,  244.04574585,  211.51608276,  -40.39221573,
        -149.74568176]], dtype=float32)
```